crates/topos-tce-storage/README.md

```diff
@@ -0,0 +1,81 @@
+
+- The [`EpochValidatorsStore`](struct@epoch::EpochValidatorsStore) is responsible for managing the list of validators for each `epoch`.
+- The [`FullNodeStore`](struct@fullnode::FullNodeStore) is responsible for managing all the persistent data such as [`Certificate`] delivered and associated `streams`.
+- The [`IndexStore`](struct@index::IndexStore) is responsible for managing indexes in order to collect information about the broadcast and the network.
+- The [`ValidatorStore`](struct@validator::ValidatorStore) is responsible for managing the pending certificates pool and all the transient and volatile data.
+
+For more information about a `store`, see the related doc.
+
+Next, we've the list of the different tables and their responsibilities:
+
+- The [`EpochValidatorsTables`](struct@epoch::EpochValidatorsTables) is responsible for storing the list of validators for each `epoch`.
+- The [`ValidatorPerpetualTables`](struct@validator::ValidatorPerpetualTables) is responsible for storing the [`Certificate`] delivered and all the persitent data related to the broadcast.
+- The [`ValidatorPendingTables`](struct@validator::ValidatorPendingTables) is responsible for storing the pending certificates pool and all the transient and volatile data.
+- The [`IndexTables`](struct@index::IndexTables) is responsible for storing indexes about the delivery of [`Certificate`] such as `target subnet stream`.
+
+### Special Considerations
+
+When using the storage layer, you need to be aware of the following:
+- The storage layer is using [rocksdb](https://rocksdb.org/) as a backend, which means that this storage doesn't need external service, as `rocksdb` is embeddable kv store.
+- The storage layer is using [`Arc`](struct@std::sync::Arc) to share the stores between threads. It also means that a `store` is only instantiated once.
+- Some functions are batching multiple writes in one transaction. But not all functions are using it.
+
+### Design Philosophy
+
+The choice of using [rocksdb](https://rocksdb.org/) as a backend was made because it is a well known and battle tested database.
+It is also very fast and efficient when it comes to write and read data.
+
+Multiple `stores` and `tables` exists in order to allow admin to deal with backups or
+snapshots as they see fit. You can pick and choose which `tables` you want to backup without having to backup the whole database.
+
+By splitting the data in dedicated tables we define strong separation of concern
+directly in our storage.
+
+`RocksDB` is however not the best fit when it comes to compose or filter data based on the data
+itself.
+
+For complex queries, another database like [`PostgreSQL`](https://www.postgresql.org/) or [`CockroachDB`](https://www.cockroachlabs.com/) could be used as a Storage for projections.
+The source of truth would still be [rocksdb](https://rocksdb.org/) but the projections would be stored in a relational database, allowing for more complex queries.
```

Comment:
I think it does, a projection database is not meant to be the SSOT of any data. It can be deleted and rebuilt from scratch without issues, and losing the projection database should lead to any data loss.

The point here is to explain why some capabilities, like having complex queries, are not meant to be part of the different stores that rely on `RocksDB`. 
It also points that we could provide a dedicated relational database layer implementation in order to provide those queries for the different API.

It is mainly here to prevent people asking or searching how to do things that is not possible for those implementations.

Reply:
I have to say I agree with @dvdplm here. This is not a paper which discusses possibilities. The target audience is people who want to contribute to the project, and want a quick read on what piece of code means what. 

Mentioning other solutions can lead to being outdated faster, discussions around it etc. etc. 

To have this information is valid, I just don't think docs in a repository are the place to have it. One example: They might not have this question but another idea. Like how many should we guess and what should we cover? 

We should just cover what we have. Every discussions or anything else, people can open an issue on GitHub and we can discuss it there.

Reply:
Comments like this is exactly why we shouldn't argue for or against a solution in documentation. 
https://github.com/topos-protocol/topos/pull/353/files#r1383286732

Because it never ends :) There is always something not accurately portrait etc.

Reply:
Ideally/Eventually we could refer to the RFC that introduced this decision which I think explained the rationales and the alternatives, although the current RFC is bit old and outdated on some parts (and also the repo is still private)

In the meantime, I'm good with either removing this part, or squashing by having not more than an one-liner e.g., "Our state representation along with the ways we are accessing it makes RocksDB pretty suitable in our context."